windows-nsis/openvpn.nsi

```diff
@@ -175,6 +175,89 @@ ReserveFile "install-whirl.bmp"
 	Pop $R0
 !macroend
 
+Function CacheServiceState
+	; We will set the defaults a service only if it did not exist before:
+	; otherwise we restore the previous state. We cache startuptype only for
+	; OpenVPNService because we need to reinstall it. Reinstall is required
+	; as OpenVPNService might be pointing to the old legacy service.
+	Var /GLOBAL iservice_existed
+	Var /GLOBAL iservice_was_running
+	Var /GLOBAL legacy_service_existed
+	Var /GLOBAL legacy_service_was_running
+	Var /GLOBAL service_existed
+	Var /GLOBAL service_starttype
+	Var /GLOBAL service_was_running
+
+	DetailPrint "Caching service states"
+
+	SimpleSC::ExistsService "OpenVPNServiceInteractive"
+	Pop $iservice_existed
+	SimpleSC::GetServiceStatus "OpenVPNServiceInteractive"
+	Pop $0
+	Pop $iservice_was_running
+
+	SimpleSC::ExistsService "OpenVPNServiceLegacy"
+	Pop $legacy_service_existed
+	SimpleSC::GetServiceStatus "OpenVPNServiceLegacy"
+	Pop $0
+	Pop $legacy_service_was_running
+
+	SimpleSC::ExistsService "OpenVPNService"
+	Pop $service_existed
+	SimpleSC::GetServiceStartType "OpenVPNService"
+	Pop $0
+	Pop $service_starttype
+	SimpleSC::GetServiceStatus "OpenVPNService"
+	Pop $0
+	Pop $service_was_running
+FunctionEnd
+
+Function RestoreServiceState
+	; Use default settings for services which were not installed before
+	${IfNot} $iservice_existed == 0
+		DetailPrint "Enabling and starting OpenVPN Interactive service" 
+		SimpleSC::SetServiceStartType "OpenVPNServiceInteractive" 2
+		SimpleSC::StartService "OpenVPNServiceInteractive" "" 5
+	${EndIf}
+
+	${IfNot} $legacy_service_existed == 0
+		DetailPrint "Setting starttype of OpenVPN Legacy Service to manual"
+		SimpleSC::SetServiceStartType "OpenVPNServiceLegacy" 3
```

Comment:
Same here: if not existed, starttype is already set during install. But reinstall to new location requires a binary_path reset.